js/widgets/navbar.js

```diff
@@ -35,59 +35,112 @@ return $.widget( "mobile.navbar", {
 
 	options: {
 		iconpos: "top",
-		grid: null
+		maxbutton: 5
 	},
 
 	_create: function() {
 
-		var $navbar = this.element,
-			$navbuttons = $navbar.find( "a, button" ),
-			iconpos = $navbuttons.filter( ":jqmData(icon)" ).length ? this.options.iconpos : undefined;
+		var self = this,
+			navbar = self.element,
+			navButtons = navbar.find( "a" ),
+			numButtons = navButtons.length,
+			maxButton = this.options.maxbutton,
+			iconpos = navButtons.filter( ":jqmData(icon)" ).length ?
+				self.options.iconpos : undefined;
 
-		$navbar.addClass( "ui-navbar" )
+		navbar.addClass( "ui-navbar" )
 			.attr( "role", "navigation" )
 			.find( "ul" )
-				.jqmEnhanceable()
-				.grid( { grid: this.options.grid } );
-
-		$navbuttons
-			.each( function() {
-				var icon = $.mobile.getAttribute( this, "icon" ),
-					theme = $.mobile.getAttribute( this, "theme" ),
-					classes = "ui-button";
-
-				if ( theme ) {
-					classes += " ui-button-" + theme;
-				}
-				if ( icon ) {
-					classes += " ui-icon-" + icon + " ui-button-icon-" + iconpos;
-				}
-				$( this ).addClass( classes );
+			.jqmEnhanceable();
+
+		$.extend( self, {
```

Comment:
no need to extend just do `this.{{thing}} = bar;` no need to merge anything here